test/functional/lua/buffer_updates_spec.lua

```diff
@@ -1034,6 +1039,48 @@ describe('lua: nvim_buf_attach on_bytes', function()
       }
     end)
 
+    local function do_test_issue_12861(mode)
```

Comment:
I would call this fuction `test_lockmarks` (even as we run it without lockmarks, the purpose is still to verify conistency of lockmarks with non-lockmarks).

Reply:
Done